com.lunarclient.LunarClient.yml

```diff
@@ -0,0 +1,71 @@
+app-id: com.lunarclient.LunarClient
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+command: lunarclient
+separate-locales: false
+rename-icon: lunarclient
+finish-args:
+  - --persist=.minecraft
+  - --persist=.lunarclient
+  - --share=network
+  - --device=dri
+  - --socket=x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --persist=.java
+modules:
+  - name: unappimage
+    buildsystem: simple
+    build-commands:
+      - make -C squashfs-tools install INSTALL_DIR=/app/bin
+    sources:
+      - type: git
+        url: https://github.com/refi64/unappimage
+        commit: d7f86f2a0d7ec3a69211125207d5f127386b849a
+
+  - name: lunarclient
+    buildsystem: simple
+    cleanup:
+      - squashfs-root
+    sources:
+      - type: extra-data
+        url: https://launcherupdates.lunarclientcdn.com/Lunar%20Client-2.9.4.AppImage
+        size: 87798866 
+        sha256: b1509d4b0fe32ffd1d37720ec2d85f983bcb6ad5b3c40421a6a2fc7e9f227a17
+        filename: lunar-client.appimage
+        x-checker-data:
+            type: rotating-url
```

Comment:
Have you tested it that it actually works? Looking at the example in the README and usage on Flathub at https://github.com/search?q=org%3Aflathub+rotating-url&type=code, I sincerely doubt it.

Reply:
Use the electron checker. This won't work

Reply:
Are you sure about the electron checker?

Reply:
Yes. It's made specifically for apps like this

Rotating url only checks for a link redirecting to the latest version, this isn't the case here

Reply:
Any luck fixing this?

Reply:
Yes, if author switches to electron checker

Reply:
Which is the point of my question.

Reply:
Running
```yml
url: https://launcherupdates.lunarclientcdn.com/Lunar%20Client-2.9.4.AppImage
        size: 87798866 
        sha256: de1d8516920846d6ba844b1afc7d31a3cf8352f85d43206ff5da9d77fbe5c19e
        filename: lunar-client.appimage
        x-checker-data:
            type: electron-updater
```

Does in fact provide use with an updated URL
```
State.LATEST|BROKEN: lunar-client.appimage
 Has a new version:
  URL:       https://launcherupdates.lunarclientcdn.com/Lunar%20Client-2.10.1.AppImage
  MD5:       d4eef58ee5758ae081af0c3f35ecccf7
  SHA1:      2b350cb2d0bfd2e8b2e0a907c956935331dae6cd
  SHA256:    de1d8516920846d6ba844b1afc7d31a3cf8352f85d43206ff5da9d77fbe5c19e
  SHA512:    77b39e542c00ce9ebad9d2921a6c8f35314a025333c6607703f7308756d81df82bda89b53f754e44ed8329e6ea3c9ee5d79e28a7db49ac869efdecbd36b846dc
  Size:      87774312
  Version:   2.10.1
  Timestamp: 2022-04-07 19:48:25.342000+00:00
  ```
  
  But change it to 2.10.1, and it will *still* pump out that there is an update. @rbrick Do you think you can educate us on how to let flatpak check for an update, so it can automatically update the launcher? A url or anything that we can use?

Reply:
I am pretty sure it does that because you didn't copy the sha256 in updated one. I had that issue. What if you let it update it not only check?

Reply:
I just double tested it, Same result. Try it yourself too.

It keeps spitting out that there is an update

Reply:
![image](https://user-images.githubusercontent.com/8540562/172088922-f1263421-3745-47d7-bcce-5aa3bc239af9.png)

You might be forgetting to update the URL? I have not been able to reproduce any issues when changing both the URL and sha256 hahs.

Reply:
Following @DaPigGuy pointer, I have realized the error of my ways.

The FEDC does work